### ISSUE TITLE
[Bug]: Cross Origin Isolation not enabled

### ISSUE THREAD

**Issue by fabiospampinato** [electron/electron#35905]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

21.0.0

### What operating system are you using?

macOS

### Operating System Version

macOS Monterey 12.5.1

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

It should be possible to enable Cross Origin Isolation, as some APIs, like `process.measureUserAgentSpecificMemory`, are unavailable otherwise.

### Actual Behavior

It doesn't seem possible to enable it, at least not in any way that I've been able to discover.

Basically following [this](https://web.dev/cross-origin-isolation-guide/#enable-cross-origin-isolation) guide doesn't result in cross origin isolation being enabled.

The magic headers seem set properly:

```ts
  mainWindow.loadURL(`file://${path.join(__dirname, 'index.html')}`, {
    extraHeaders: 'Cross-Origin-Opener-Policy: same-origin\nCross-Origin-Embedder-Policy: require-corp'
  })
```

But nothing changed:

```ts
console.log(globalThis.crossOriginIsolated) // => false
```

### Testcase Gist URL

https://gist.github.com/d98f83f2a1e6b3a62ad0a85a529719f4

**Comment by nornagon** [comment]
This is also Chrome's behavior, so I don't think this is an Electron issue.

![Screen Shot 2022-10-04 at 9 41 05 AM](https://user-images.githubusercontent.com/172800/193877474-49d07d04-6e9b-4226-8afc-fac4d5ada5e8.png)

**Comment by fabiospampinato** [comment]
But I'm explicitly passing the required headers. So how am I supposed to get this to work? Do I need a temporary server on the main process that just serves that one html file?

Like what's the recommended way for accessing APIs such as `performance.measureUserAgentSpecificMemory`?

**Comment by 0xdevalias** [comment]
See also:

- https://github.com/electron/electron/issues/37147

**Comment by 0xdevalias** [comment]
@nornagon I think you were sort of missing the point of this when you closed it.

Relevant context from the other issue where this originated from:

> NB https://github.com/WICG/performance-measure-memory seems relevant. Enable with `enableBlinkFeatures: 'MeasureMemory'`. Does that solve your problem?
> 
> _Originally posted by @nornagon in https://github.com/electron/electron/issues/30383#issuecomment-892194040_

> @nornagon I think I need to set some CORS policies or something to get that working, I'm not quite sure how to do that yet, I'll update you once I can figure that out.
> 
> _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-893775990_

> @nornagon Sorry for the ping, I can't find any resource on how to make the renderer process cross origin isolated, which my understanding is is a requirement for being able to use this new API. Do you know what the solution here is?
> 
> _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-1267079565_

> > how to make the renderer process cross origin isolated
> 
> renderer processes are isolated per origin by default. how are you managing to make them _not_ cross origin isolated?
> 
> _Originally posted by @nornagon in https://github.com/electron/electron/issues/30383#issuecomment-1267236122_

> @nornagon I've opened an issue about this [here](https://github.com/electron/electron/issues/35905). I'm getting `globalThis.crossOriginIsolated === false` in a plain fiddle using Electron v21 🤷‍♂️
> 
> _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-1267249805_
